### PR TITLE
Responsive box

### PR DESCRIPTION
**Pull request by juanfabrega**

### CODE REVIEW

File: src/components/Box/Box.tsx [hunk 1]
```diff
@@ -27,33 +57,17 @@ export interface BoxProps {
   /**
    * How to align the contents along the cross axis.
    */
-  align?:
-  'start'
-  | 'end'
-  | 'center'
-  | 'baseline'
-  | 'stretch';
+  alignItems?: CssAlign | ResponsiveGeneric;
```
Comment (nathanyoung) [inline src/components/Box/Box.tsx:60]: story needs to be updated to reflect new prop name and possible values

File: src/components/Box/Box.tsx [hunk 2]
```diff
@@ -27,33 +57,17 @@ export interface BoxProps {
...
    * This property has no effect when there is only one line of flex items.
    */
-  alignContent?:
-  'start'
-  | 'end'
-  | 'center'
-  | 'stretch'
-  | 'between'
-  | 'around';
+  alignContent?: CssAlignContent | ResponsiveGeneric;
```
Comment (nathanyoung) [inline src/components/Box/Box.tsx:65]: story needs to be updated to reflect new prop name and possible values

File: src/components/Box/Box.tsx [hunk 3]
```diff
@@ -27,33 +57,17 @@ export interface BoxProps {
...
    * How to align along the cross axis when contained in a Box.
    * This allows the default alignment (or the one specified by `align`) to be overridden for the individual Box.
    */
-  alignSelf?:
-  'start'
-  | 'end'
-  | 'center'
-  | 'baseline'
-  | 'stretch';
+  alignSelf?: CssAlign | ResponsiveGeneric;
```
Comment (nathanyoung) [inline src/components/Box/Box.tsx:70]: story needs to be updated to reflect new prop name and possible values

File: src/components/Box/Box.tsx [hunk 4]
```diff
@@ -82,95 +96,78 @@ export interface BoxProps {
...
    */
-  justify?:
-  'around'
-  | 'between'
-  | 'center'
-  | 'end'
-  | 'evenly'
-  | 'start'
-  | 'stretch';
+  justifyContent?: CssJustifyContent | ResponsiveGeneric;
```
Comment (nathanyoung) [inline src/components/Box/Box.tsx:120]: story needs to be updated to reflect new prop name and possible values

**Review by nathanyoung — CHANGES_REQUESTED** [review]

**Comment by juanfabrega** [comment]
[Add responsive capability to props in the Box component](https://trello.com/c/CDvpC92G/232-add-responsive-capability-to-props-in-the-box-component)

**Comment by juanfabrega** [comment]
[Add responsive capability to props in the Box component](https://trello.com/c/CDvpC92G/232-add-responsive-capability-to-props-in-the-box-component)

**Comment by juanfabrega** [comment]
[Add responsive capability to props in the Box component](https://trello.com/c/CDvpC92G/232-add-responsive-capability-to-props-in-the-box-component)

**Review by nathanyoung — APPROVED** [review]

**Comment by nathanyoung** [comment]
https://github.com/palmetto/palmetto-components/pull/257

**Comment by nathanyoung** [comment]
https://github.com/palmetto/palmetto-components/issues/263

**Pull request merged**